gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -163,8 +165,14 @@ public MRJobLauncher(Properties jobProps, Configuration conf)
     this.parallelRunnerThreads = Integer.parseInt(jobProps.getProperty(ParallelRunner.PARALLEL_RUNNER_THREADS_KEY,
         Integer.toString(ParallelRunner.DEFAULT_PARALLEL_RUNNER_THREADS)));
 
+    // StateStore interface uses the following key (rootDir, storeName, tableName)
+    // The state store base is the root directory and the last two elements of the path are used as the storeName and
+    // tableName. Create the state store with the root at jobOutputPath. The task state will be stored at
+    // jobOutputPath/output/taskState.tst, so output will be the storeName.
+    taskStateStore = new FsStateStore<>(this.fs, jobOutputPath.toString(), TaskState.class);
```

Comment:
MR has a shared file system, so there is less of a need for non-fs state store, but I guess it doesn't hurt to have the flexibility, so I'll add the code for arbitrary store type.